Wikipedia/Code/main.m

```diff
@@ -5,14 +5,51 @@
 
 #import "AppDelegate.h"
 
+#if DEBUG
+/**
+ *  Dummy application delegate for use in unit testing.
+ *
+ *  Visual tests require that the application has a @c keyWindow, and we don't pass the regular application delegate to
+ *  prevent unintended side effects from regular application code when testing.
+ */
+@interface WMFDummyAppDelegate : UIResponder <UIApplicationDelegate>
+@property (nonatomic, strong) UIWindow* window;
+@end
+#endif
+
 int main(int argc, char* argv[]) {
+#if DEBUG
     // disable app when unit testing to allow tests to run in isolation (w/o side effects)
     BOOL const isUnitTesting = NSClassFromString(@"XCTestCase") != nil;
+#endif
     @autoreleasepool {
         return UIApplicationMain(argc,
                                  argv,
                                  nil,
-                                 isUnitTesting ? nil : NSStringFromClass([AppDelegate class]));
+                                 #if DEBUG
```

Comment:
i'll make this more straightforward w/ a variable instead of breaking up the ternary statement. is there something else about `DEBUG` usage you had issue with?